clicommand/pipeline_upload.go

```diff
@@ -236,6 +241,23 @@ var PipelineUploadCommand = cli.Command{
 			return
 		}
 
+		if len(cfg.RedactedVars) > 0 {
+			needles := redaction.GetValuesToRedact(shell.StderrLogger, cfg.RedactedVars, env.FromSlice(os.Environ()).ToMap())
+			serialisedPipeline, err := result.MarshalJSON()
+
+			if err != nil {
+				l.Fatal("Pipeline serialization of \"%s\" failed (%s)", src, err)
+			}
+
+			stringifiedserialisedPipeline := string(serialisedPipeline)
+
+			for _, needle := range needles {
+				if strings.Contains(stringifiedserialisedPipeline, needle) {
+					l.Fatal("Couldn't upload %q pipeline. Refusing to upload pipeline containing redacted vars. Ensure your pipeline does not include secret values or interpolated secret values", src)
```

Comment:
I did wonder about whether `GetValuesToRedact` could return tuples of `(env_key, env_value)` so that this error could say "Whoops your pipeline interpolated the value of the $env_key" and point folks closer to the answer. What do you think?

Reply:
Yea, a bit more specificity never hurts! 
How about this?
>Couldn't upload %q pipeline. Refusing to upload pipeline containing interpolated values of the $env_key. Ensure your pipeline does not include secret values or interpolated secret values

Reply:
Yeah I think that would be great 😄